AK/TypeCasts.h

```diff
@@ -56,7 +56,11 @@ ALWAYS_INLINE CopyConst<InputType, OutputType>* as_if(InputType& input)
 {
     if (!is<OutputType>(input))
         return nullptr;
-    return static_cast<CopyConst<InputType, OutputType>*>(&input);
+    if constexpr (IsBaseOf<InputType, OutputType>) {
+        return static_cast<CopyConst<InputType, OutputType>*>(&input);
+    } else {
+        return dynamic_cast<CopyConst<InputType, OutputType>*>(&input);
+    }
```

Comment:
I think it makes sense to keep `as` aligned with this new behavior (principle of least astonishment).

Reply:
I've rewritten `as` in terms of `as_if`, so now both `as` and `as_if` should succeed if `is` succeeds.